### PR TITLE
improvement(collect kallsyms): Collect Seastar kallsyms on boot

### PR DESCRIPTION
**Pull request by yarongilor**
The kallsyms map for kernel debug should be collected
	dynamically every node boot.

This is a complementary PR to: https://github.com/scylladb/scylla-cluster-tests/pull/5093 since backporting to perf-v12 is not possible.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by yarongilor** [comment]
Have to verify a performance test with any reboot nemesis passed ok.

**Comment by yarongilor** [comment]
Looks like the test pass ok. It failed after 4 hours due to spot termination (missed 2 hours to fully complete).
db collected logs had:
```
yarongilor@yaron-pc:~/Downloads/logs/db-cluster-e6282912$ ll
total 9640
drwxr-xr-x   5 yarongilor yarongilor    4096 Aug 30 14:16 ./
drwxrwxr-x 100 yarongilor yarongilor   16384 Aug 30 14:52 ../
-rw-rw-r--   1 yarongilor yarongilor 1391298 Aug 30 14:16 kallsyms_20220830_071052.tar.gz
-rw-rw-r--   1 yarongilor yarongilor 1391061 Aug 30 14:16 kallsyms_20220830_071053.tar.gz
-rw-r--r--   1 yarongilor yarongilor 7053296 Aug 30 14:16 messages.log
drwxr-xr-x   2 yarongilor yarongilor    4096 Aug 30 14:16 perf-latency-nemesis-kallsyms-db-node-e6282912-1/
drwxr-xr-x   2 yarongilor yarongilor    4096 Aug 30 14:16 perf-latency-nemesis-kallsyms-db-node-e6282912-2/
drwxr-xr-x   2 yarongilor yarongilor    4096 Aug 30 14:16 perf-latency-nemesis-kallsyms-db-node-e6282912-3/
yarongilor@yaron-pc:~/Downloads/logs/db-cluster-e6282912$ ll perf-latency-nemesis-kallsyms-db-node-e6282912-1/kallsyms_20220830_071053.tar.gz 
-rw-rw-r-- 1 yarongilor yarongilor 1391023 Aug 30 14:16 perf-latency-nemesis-kallsyms-db-node-e6282912-1/kallsyms_20220830_071053.tar.gz
yarongilor@yaron-pc:~/Downloads/logs/db-cluster-e6282912$ ll perf-latency-nemesis-kallsyms-db-node-e6282912-2/kallsyms_20220830_071052.tar.gz 
-rw-rw-r-- 1 yarongilor yarongilor 1391298 Aug 30 14:16 perf-latency-nemesis-kallsyms-db-node-e6282912-2/kallsyms_20220830_071052.tar.gz
yarongilor@yaron-pc:~/Downloads/logs/db-cluster-e6282912$ ll perf-latency-nemesis-kallsyms-db-node-e6282912-3/kallsyms_20220830_071053.tar.gz 
-rw-rw-r-- 1 yarongilor yarongilor 1391061 Aug 30 14:16 perf-latency-nemesis-kallsyms-db-node-e6282912-3/kallsyms_20220830_071053.tar.gz
```

**Comment by fruch** [comment]
linting and unittests are failing.


Can't merge it like that

**Comment by fgelcer** [comment]
i'm re-running it on https://jenkins.scylladb.com/job/scylla-staging/job/Longevity_yaron/job/scylla-master-perf-regression-latency-250gb-with-nemesis-kallsyms/4/

as we didn't see the most relevant part, that is the new `kallsyms` collected for new nodes...
also, IDK why, but the files are into `.tar.gz` files, but i think it was a feature added to later branches, where we stopped `tar'ing` files inside `tar` files

**Comment by yarongilor** [comment]
> linting and unittests are failing.
> 
> Can't merge it like that

@fruch , as i mentioned above, list and unit-tests fail on non related issues - 
```
10:12:25  ************* Module sdcm.nemesis
10:12:25  sdcm/nemesis.py:75:0: C0411: third party import "from argus.db.db_types import NemesisStatus, NemesisRunInfo, NodeDescription" should be placed before "from sdcm.paths import SCYLLA_YAML_PATH" (wrong-import-order)
10:12:25 



10:18:15  >       assert event.timestamp == 1624895582.269804
10:18:15  E       AssertionError: assert 1656431582.269804 == 1624895582.269804
10:18:15  E         +1656431582.269804
10:18:15  E         -1624895582.269804

```
The branch contains these errors in the first place.

**Review by fgelcer — APPROVED** [review]
> i'm re-running it on https://jenkins.scylladb.com/job/scylla-staging/job/Longevity_yaron/job/scylla-master-perf-regression-latency-250gb-with-nemesis-kallsyms/4/
> 
> as we didn't see the most relevant part, that is the new `kallsyms` collected for new nodes... also, IDK why, but the files are into `.tar.gz` files, but i think it was a feature added to later branches, where we stopped `tar'ing` files inside `tar` files

there is an issue with the files collected outside the node's folder, but this is something we want to solve in the future, hence approving this PR
* the issue is that `node-2` and `node-3` have the exact same filename, so in the root folder, they appear to have 1 missing file, but for each node's folder, the files are in there, hence we are safe.
LGTM

**Comment by fgelcer** [comment]
@scylladb/qa-maintainers ping

**Comment by soyacz** [comment]
Why are unit tests / precommit test failing?

**Comment by fgelcer** [comment]
closing, as we will be creating soon a new branch for the performance jobs

**Pull request closed without merging**
